python/cuml/cuml/internals/base.pyx

```diff
@@ -988,3 +994,27 @@ class UniversalBase(Base):
         estimator.output_mem_type = MemoryType.host
 
         return estimator
+
+    def get_params(self, deep=True):
+        """
+        If accelerator is active, we return the params of the CPU estimator
+        being helf by the class, otherwise we just call the regular
+        get_params of the Base class.
+        """
+        if GlobalSettings().accelerator_active or self._experimental_dispatching:
+            return self._cpu_hyperparams_dict
+        else:
+            return super().get_params(deep=deep)
+
+    def set_params(self, **params):
+        """
+        For setting parameters, when the accelerator is active, we translate
+        the parameters to set the GPU params, and also update the
+        params of the CPU class. Otherwise dispatching to the CPU class after
+        updating params of the GPU estimator will dispatch to an estimator
+        with outdated params.
+        """
+        self._cpu_hyperparams_dict.update(params)
```

Comment:
This call will fail with an `AttributeError` when `_cpu_hyperparams_dict` was not initialized yet.

The issue is that this new attribute is only initialized within the `ProxyEstimator` class, but used within the `UniversalBase` class. We should probably initialize it within the base class.